web3.js/src/transaction.ts

```diff
@@ -752,9 +800,11 @@ export class Transaction {
   static populate(
     message: Message,
     signatures: Array<string> = [],
+    lastValidBlockHeight?: number,
```

Comment:
Wait, I'm confused. The `Message` type can have `latestBlockhash` on it. If present, then you have all you need to reconstruct a `Transaction` with a `lastValidBlockHash`. Instead of changing the method sig here, can you just inspect message, looking for `latestBlockhash.lastValidBlockHeight`, and on the `compileMessage` side make sure that you write out a `latestBlockhash`?

Reply:
That was part of this PR, but we didn't finish it because we realized it was going to be more involved. If you look down further in `Message` to the `from` method, we don't do anything with `lastValidBlockHeight`. So if we include `latestBlockhash` in a constructed message, `lastValidBlockHeight` is `undefined`.